torch/csrc/jit/tensorexpr/cuda_half_support.h

```diff
@@ -50,8 +74,16 @@ class CudaHalfChecker : public IRMutator {
     // just don't allow half casts we didn't insert.
     if (v->dtype().scalar_type() == ScalarType::Half) {
       if (inserted_half_casts_.count(v) < 1) {
-        // TODO: discards lanes.
-        return new Cast(kFloat, child);
```

Comment:
Reason for this change ? Retaining cast to float seemed more semantics-preserving